extensions/wikia/Recirculation/js/recirculation.js

```diff
@@ -28,139 +32,78 @@ require([
 	 */
 	'use strict';
 
-	var recircModules = [
-			{
-				id: 'LI_rail',
-				viewModule: 'ext.wikia.recirculation.views.premiumRail',
-				options: {
-					max: 5,
-					widget: 'wikia-rail',
-					source: 'fandom',
-					opts: {
-						resultType: 'cross-domain',
-						domainType: 'fandom.wikia.com'
-					}
+	var $mixedContentFooter = $('#mixed-content-footer'),
+		railRecirculation = {
+			max: 5,
+			widget: 'wikia-rail',
+			modelName: 'ns',
+			opts: {
+				resultType: 'cross-domain',
+				domainType: 'fandom.wikia.com'
+			}
+		},
+		mixedContentFooter = {
+			nsItems: {
+				max: $mixedContentFooter.data('number-of-ns-articles'),
+				widget: 'wikia-impactfooter',
+				modelName: 'ns',
+				opts: {
+					resultType: 'cross-domain',
+					domainType: 'fandom.wikia.com'
 				}
 			},
-			// TODO - remove old recirculation footers
-			// {
-			// 	id: 'LI_impactFooter',
-			// 	viewModule: 'ext.wikia.recirculation.views.impactFooter',
-			// 	options: {
-			// 		max: 9,
-			// 		widget: 'wikia-impactfooter',
-			// 		source: 'fandom',
-			// 		opts: {
-			// 			resultType: 'cross-domain',
-			// 			domainType: 'fandom.wikia.com'
-			// 		}
-			// 	}
-			// },
-			// {
-			// 	id: 'LI_footer',
-			// 	viewModule: 'ext.wikia.recirculation.views.footer',
-			// 	options: {
-			// 		max: 3,
-			// 		widget: 'wikia-footer-wiki-rec',
-			// 		source: 'wiki',
-			// 		title: 'Discover New Wikis',
-			// 		width: 332,
-			// 		height: 187,
-			// 		flush: true,
-			// 		opts: {
-			// 			resultType: 'subdomain',
-			// 			domainType: 'fandom.wikia.com'
-			// 		}
-			// 	}
-			// }
-		],
-		logGroup = 'ext.wikia.recirculation.experiments.mix',
-		// Each view holds an array of promises used to gather data for that view
-		views = {},
-		// An array of promises to keep track of which views have completed rendering
-		completed = [],
-		liftigniterHelpers = {};
+			wikiItems: {
+				max: $mixedContentFooter.data('number-of-wiki-articles'),
+				widget: 'wikia-footer-wiki-rec',
+				modelName: 'wiki',
+				opts: {
+					domainType: 'fandom.wikia.com'
+				}
+			}
+		},
+		logGroup = 'ext.wikia.recirculation.experiments.mix';
 
-	if (w.wgContentLanguage === 'en') {
+	if (w.wgContentLanguage !== 'en') {
 		if (videosModule) {
 			videosModule('#recirculation-rail');
 		}
-		discussions();
+		// oldDiscussions();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)